### PR TITLE
sdlpop 1.22

### PR DESCRIPTION
**Pull request by chenrui333**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by chenrui333** [comment]
this should be a cask formula?

```
if (NOT(APPLE AND CREATE_BUNDLE))
    add_executable(prince ${SOURCE_FILES})
else()

    # macOS-specific:
    # By default, a normal binary executable will be created.
    # If you want an Application Bundle instead, pass CREATE_BUNDLE to CMake like so: -DCREATE_BUNDLE=1

    # NOTE: Currently, the SDL2 and SDL2_image frameworks are not being correctly included/linked in the bundle!
    #       As long as this isn't fixed, these bundles will only work if SDL2 and SDL2_image are installed.

    set (SDLPoP_VERSION "1.22")
    set (MACOSX_BUNDLE_INFO_STRING          ${PROJECT_NAME})
    set (MACOSX_BUNDLE_ICON_FILE            "icon.icns")
    set_source_files_properties(${MACOSX_BUNDLE_ICON_FILE} PROPERTIES MACOSX_PACKAGE_LOCATION Resources)
    set (MACOSX_BUNDLE_GUI_IDENTIFIER       "org.princed.SDLPoP")
    set (MACOSX_BUNDLE_LONG_VERSION_STRING  ${SDLPoP_VERSION})
    set (MACOSX_BUNDLE_BUNDLE_NAME          "SDLPoP")
    set (MACOSX_BUNDLE_SHORT_VERSION_STRING ${SDLPoP_VERSION})
    set (MACOSX_BUNDLE_BUNDLE_VERSION       ${SDLPoP_VERSION})
    set (MACOSX_BUNDLE_COPYRIGHT            "GNU General Public Licence, v3")
    add_executable(prince MACOSX_BUNDLE ${SOURCE_FILES} ${MACOSX_BUNDLE_ICON_FILE})
    add_custom_command(TARGET prince POST_BUILD
            COMMAND ${CMAKE_COMMAND} -E copy_directory
            ${CMAKE_SOURCE_DIR}/../data ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/prince.app/Contents/MacOS/data)
    add_custom_command(TARGET prince POST_BUILD
            COMMAND ${CMAKE_COMMAND} -E copy_directory
            ${CMAKE_SOURCE_DIR}/../doc ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/prince.app/Contents/MacOS/doc)
    add_custom_command(TARGET prince POST_BUILD
            COMMAND ${CMAKE_COMMAND} -E copy
            ${CMAKE_SOURCE_DIR}/../SDLPoP.ini ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/prince.app/Contents/MacOS/SDLPoP.ini)
endif()

```

**Comment by cho-m** [comment]
The failure may be related to https://github.com/NagyD/SDLPoP/commit/3b83eb62d473d17eac4544e9c1e62a88b91ef92f where the header was changed to path shown by `pkg-config sdl2 --cflags`; however, the [Makefile](https://github.com/NagyD/SDLPoP/blob/v1.22/src/Makefile) has macOS-specific logic 

---

Not sure about if better as Cask. It is a video game with pre-built DMGs available https://www.popot.org/get_the_games.php?game=SDLPoP

**Comment by carlocab** [comment]
Might be better to use a CMake build since the `Makefile` seems to assume you're using MacPorts: https://github.com/NagyD/SDLPoP/blob/431e024ead57d035e65af44152fbac6974f76d94/src/Makefile#L15-L17

We can always pass `-DCREATE_BUNDLE=OFF` to avoid creating an app bundle.

**Comment by alebcay** [comment]
Well, I just had a chance to look at the CMake build. It assumes that `sdl2` and `sdl2_image` live in the same directory (we could possibly pass `HOMEBREW_PREFIX/include/SDL2` and `HOMEBREW_PREFIX/lib` but this feels a bit unconventional).

Considering that patching either the CMake or Make builds are necessary, I'd prefer to stick with the Makefile here. It's easier to inject compiler flags and I think the diff will be smaller.

**Comment by alebcay** [comment]
It looks like the CMake build is broken too:
```
if (NOT(WIN32) AND (DEFINED SDL2))
    include_directories(${SDL2}/include)
    link_directories(${SDL2}/lib)
endif()
```

Those header files will never get found because they live at `#{include}/SDL2/SDL.H` etc.

**Comment by cho-m** [comment]
The easiest patch to try is just deleting the macOS-specific logic from Makefile:
```diff
diff --git a/src/Makefile b/src/Makefile
index f7181ef..a4deeb2 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -12,14 +12,8 @@ OS      := $(shell uname)
 CPPFLAGS += -Wall -D_GNU_SOURCE=1
 CFLAGS += -std=gnu99 -O2
 
-ifeq ($(OS),Darwin)
-LIBS := $(shell sdl2-config --libs) -lSDL2_image
-CFLAGS += -I/opt/local/include
-CPPFLAGS += -D_THREAD_SAFE -DOSX
-else
 LIBS := $(shell pkg-config --libs   sdl2 SDL2_image)
 CFLAGS += $(shell pkg-config --cflags sdl2 SDL2_image)
-endif
 
 all: $(BIN)
 
```

File: Formula/sdlpop.rb [hunk 1]
```diff
@@ -41,3 +44,23 @@ def caveats
...
+--- a/src/Makefile
++++ b/src/Makefile
+@@ -12,14 +12,8 @@ OS      := $(shell uname)
+ CPPFLAGS += -Wall -D_GNU_SOURCE=1
+ CFLAGS += -std=gnu99 -O2
+ 
+-ifeq ($(OS),Darwin)
+-LIBS := $(shell sdl2-config --libs) -lSDL2_image
+-CFLAGS += -I/opt/local/include
+-CPPFLAGS += -D_THREAD_SAFE -DOSX
```
Comment (carlocab) [inline Formula/sdlpop.rb:60]: I think we might want to keep these `CPPFLAGS`.

This also looks like a patch that will never get upstreamed. We should turn it into one that can be.

Comment (cho-m) [reply Formula/sdlpop.rb:60]: `CFLAGS += $(shell pkg-config --cflags sdl2 SDL2_image)` will add `-D_THREAD_SAFE` to `CFLAGS`. Not sure if `-DOSX` will impact build.

Assuming MacPorts properly includes the `pkg-config` files for SDL2, I think the upstream should be able to do something similar to this patch.

Here is output of `pkg-config` on my system:
```console
❯ pkg-config --cflags sdl2 SDL2_image
-D_THREAD_SAFE -I/opt/homebrew/include/SDL2

❯ pkg-config --libs sdl2 SDL2_image
-L/opt/homebrew/lib -lSDL2_image -lSDL2
```

**Comment by alebcay** [comment]
Submitted PR for upstream patch 🤞 - https://github.com/NagyD/SDLPoP/pull/249.

**Comment by NagyD** [comment]
> Also, it looks like there is no need for `sdl2_mixer` anymore, 

That's correct. We have dropped it back when we added MIDI support.
Details: https://forum.princed.org/viewtopic.php?p=32920#p32920

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**